external-crates/move/move-compiler/src/diagnostics/codes.rs

```diff
@@ -182,6 +182,8 @@ codes!(
         InvalidSpecBlockMember: { msg: "invalid spec block member", severity: NonblockingError },
         InvalidRestrictedIdentifier:
             { msg: "invalid identifier escape", severity: NonblockingError },
+        NeedsTypeAnnotation:
+            { msg: "Requires a type annotation 'name: type ='", severity: BlockingError },
```

Comment:
I think we should probably just keep this as `UnexpectedToken` but just with a better error message. The codes are broad categories generally, and not normally this specific.